lib/ecto/migration/schema_migration.ex

```diff
@@ -17,15 +17,15 @@ defmodule Ecto.Migration.SchemaMigration do
   def ensure_schema_migrations_table!(repo, opts) do
     table_name = repo |> get_source |> String.to_atom()
     table = %Ecto.Migration.Table{name: table_name, prefix: opts[:prefix]}
-    meta = Ecto.Adapter.lookup_meta(repo.get_dynamic_repo())
+    meta = Ecto.Adapter.lookup_meta(get_migration_repo(repo).get_dynamic_repo())
```

Comment:
Please store it in a var so we don't need to compute it again below.

Reply:
👍  ✅  fixed